scripts/build.sh

```diff
@@ -1,43 +1,28 @@
-#!/usr/bin/env bash
+#!/bin/bash
 
-mkdir -p dist/
-mkdir -p dist-es5-module/
+set -e # exit when error
 
-rm -rf dist/*
-rm -rf dist-es5-module/*
+rm -rf dist dist-es5-module dist-es6-module connectors.js instantsearch.js widgets.js
+mkdir -p dist dist-es5-module dist-es6-module
 
-set -ev # exit when error
+echo "➡️  Bundle instantsearch.js to UMD build './dist' via webpack"
+NODE_ENV=production BABEL_ENV=production webpack --config scripts/webpack.config.js --hide-modules
 
-ROOT=`dirname "$0"`/..
+printf "dist/instantsearch.min.js gzipped will weight `cat dist/instantsearch.min.js | gzip -9 | wc -c | pretty-bytes`\n\n"
 
-license="/*! instantsearch.js ${VERSION:-UNRELEASED} | © Algolia Inc. and other contributors; Licensed MIT | github.com/algolia/instantsearch.js */"
+echo "➡️  Bundle instantsearch.js to ES5 build './dist-es5-module' via babel-cli"
+BABEL_ENV=production babel -q index.js -o dist-es5-module/index.js &
+BABEL_ENV=production babel src -d dist-es5-module/src --ignore *-test.js --quiet
 
-bundles=( 'instantsearch' )
+wait
 
-# build for jsdelivr, with everything inlined while using `preact` instead of `react`
-webpack --config scripts/webpack.config.jsdelivr.babel.js
+echo "➡️  Bundle instantsearch.js to ES6 build './dist-es6-module' via babel-cli"
+BABEL_ENV="production-es6" babel -q index.es6.js -o instantsearch.js &
+BABEL_ENV="production-es6" babel src -d dist-es6-module --ignore *-test.js --quiet &
 
-# only transpile to ES5 for package.json main entry
-babel -q index.js -o dist-es5-module/index.js
-declare -a sources=("src")
-for source in "${sources[@]}"
-do
-  babel -q --out-dir dist-es5-module/${source} ${source}
-done
+# * allow `import {connectXXX} from 'instantsearch.js/connectors'`
+# * allow `import {searchBox} from 'instantsearch.js/widgets'`
+echo "export * from './dist-es6-module/connectors/index';" >> ./connectors.js &
```

Comment:
I think it would be better if they were published because here it's like they exist but only during the build and they are hardcoded in a script which is even weirder to me.